### ISSUE TITLE
Filter examples & documentation are broken or deprecated

### ISSUE THREAD

**Issue by DamnedElric** [ical4j/ical4j#607]
**Describe the bug**

ical4j.org gives the following example for Filtering:

```java
Period period = new Period(new DateTime(today.getTime()), new Dur(1, 0, 0, 0));
Filter filter = new Filter(new PeriodRule(period));
```

Unfortunately, Filter is deprecated. The Javadoc suggests that "Most of the functionality of this class can now be achieved via lambda expressions", but there doesn't seem to be much in the way of documentation on how that would work.

There are some examples in [this blog post](https://ical4j.github.io/2021/07/28/filter-expressions.html), but those don't compile on 3.2.7

```java
Predicate<VEvent> filter = FilterExpression.parse("dtstart > now() and dtstart <= endOfDay(+P1W)").toComponentPredicate();
List<VEvent> events = calendar.getComponents("vevent").stream().filter(filter).collect(Collectors.toList());
```

`toComponentPredicate()` does not seem to exist.

It seems like FilterExpressions are intended as a replacement for the old Filters, but none of the relevant classes seem to contain any Javadoc, which makes me think that this intended as an internal API?

It would be nice if we could get a couple of working examples on the (extremely useful!) filtering. I'm willing to help out in that regard, but I'm going to need a few pointers first.

Thanks!

**Comment by benfortuna** [comment]
Yes, unfortunately the documentation is incomplete/outdated for filtering. The main docs should be here but yet to be completed: https://www.ical4j.org/filtering/

You can find some examples of filtering in the unit tests: https://github.com/ical4j/ical4j/blob/develop/src/test/groovy/net/fortuna/ical4j/filter/ComponentFilterTest.groovy

The new `ComponentFilter` and `PropertyFilter` classes replace the deprecated `Filter`, and now are used to construct a Java `Predicate` used to filter collections, etc.

I try to update the examples in the docs soon to provide some common use-cases.
